File: algorithm/searching/linearSearch/linearSearch.py
'''
This is the simple implementation of linear search algorithm in python3


'''

def linearSearch(array, size, target):
	for i in range(size):
		if array[i] == target:
			return i
	return -1

if __name__ == "__main__":
	array = [12,34,87,2,83,35,98]
	target1 = 2
	target2 = 30
	print(f"array is {array}")
	print(f"Target 1 is {target1}\nTarget 2 is {target2}") 
	print(f"Result for {target1}: {linearSearch(array, len(array),target1)}\nResult for {target2}: {linearSearch(array, len(array),target2)}\n")

File: algorithm/searching/linearSearch/linearSearch.c
#include <stdio.h>

int linearSearch(int array[], int size, int target){
	int i ;
	for(i = 0 ; i < size ; i++){
		if(array[i] == target){
			return i;	
		}
		
	}
	return -1;

}

int main(){
	int array[] = {192,54,53,46,43,76};
	int t1 = 54, t2 = 55, i;
	size_t size = (int)sizeof(array)/sizeof(array[0]);
	printf("Array is :");
	for (i=0; i<6;i++){
		printf(" %d,", array[i]);	
	}
	
	printf("Target 1: %d\nTarget 2: %d", t1, t2);
	printf("\nResult for %d : %d\nResult for %d : %d", t1, linearSearch(array, size, t1), t2, linearSearch(array, size, t2));
	
	
	return 0;

}
